### PR TITLE
Added @subchildren tag

### PR DESCRIPTION
**Pull request by imaustink**
Added expanded child menu support via @subchildren tag. Related to canjs/canjs#3266

### CODE REVIEW

File: tags/subchildren/package.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  "name": "subchildren",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC"
+}
```
Comment (chasenlehara) [inline tags/subchildren/package.json]: Is this file required by `bit-docs` for it to work?

File: templates/menu.mustache [hunk 2]
```diff
@@ -16,9 +16,7 @@
 							title="{{getLinkTitle docObject}}">
 							{{getShortTitle docObject}}
 						</a>
-						{{#if (hasOrIsCurrent docObject)}}
-							{{> menu.mustache}}
-						{{/if}}
```
Comment (chasenlehara) [inline templates/menu.mustache]: Why is the `hasOrIsCurrent` check not needed anymore?

File: templates/helpers.js [hunk 3]
```diff
@@ -402,14 +402,19 @@ DocMapInfo.prototype.getCurrentTree = function(){
...
-
     var curChildren = this.getNestedChildren(cur);
-
     this.getParents(cur, function(docObject){
         curChildren = getChildren(docObject).map(function(docObject){
             if(docObject.name === cur.name) {
+                if(cur.subchildren){
+                    curChildren = curChildren.map(function(child){
+                        if(child.docObject) child.children = self.getNestedChildren(child.docObject);
+                        return child;
```
Comment (chasenlehara) [inline templates/helpers.js]: This code was a little confusing to me because I was expecting the mapped values to be different from the input, when really this is just using `map` as `forEach`.

File: tags/subchildren/package.json (hunk 1 above)
Comment (imaustink) [inline tags/subchildren/package.json]: I thought it was but I just found an example where it doesn't use the module structure. I'ved updated the PR.

File: templates/menu.mustache (hunk 2 above)
Comment (imaustink) [inline templates/menu.mustache]: The `hasOrIsCurrent` check prevents the child menus from being printed if present but not the current page. Upstream from this `getCurrentTree` provides the tree that is being rendered by `menu.mustache`, the generated tree does not have extra children, it generates only what needs to be printed.

File: templates/helpers.js (hunk 3 above)
Comment (imaustink) [inline templates/helpers.js]: Very solid point. I've updated to a `forEach`.

File: templates/helpers.js [hunk 4]
```diff
@@ -402,14 +402,18 @@ DocMapInfo.prototype.getCurrentTree = function(){
...
     var cur = this.getCurrent();
-
     var curChildren = this.getNestedChildren(cur);
-
     this.getParents(cur, function(docObject){
         curChildren = getChildren(docObject).map(function(docObject){
             if(docObject.name === cur.name) {
+                if(cur.subchildren){
+                    curChildren.forEach(function(child, i){
+                        if(child.docObject) curChildren[i].children = self.getNestedChildren(child.docObject);
```
Comment (chasenlehara) [inline templates/helpers.js]: `curChildren[i]` is `child`, right?

Also, this line should be [split on multiple lines](https://contribute.jquery.org/style-guide/js/#spacing).

Comment (imaustink) [inline templates/helpers.js]: You're right. For some reason, I thought `forEach` made a copy of each object but that would be rediculous. Updated the PR.
